LeaderboardBackend.Test/Runs.cs

```diff
@@ -419,5 +422,239 @@ public async Task GetCategoryForRun_OK()
+
+            Run? updated = await context.FindAsync<Run>(run.Id);
+            updated!.Info.Should().Be("new info");
+            updated!.CreatedAt.Should().Be(_clock.GetCurrentInstant());
+            updated!.UpdatedAt.Should().Be(_clock.GetCurrentInstant());
+        }
+
+        [Test]
+        public async Task UpdateRun_Unauthenticated()
+        {
+            ApplicationContext context = _factory.Services.GetRequiredService<ApplicationContext>();
+
+            Run created = new()
+            {
+                CategoryId = _categoryId,
+                PlayedOn = LocalDate.MinIsoValue,
+                Time = Duration.FromSeconds(390),
+                UserId = TestInitCommonFields.Admin.Id,
+            };
+
+            context.Add(created);
+            await context.SaveChangesAsync();
+            created.Id.Should().NotBe(Guid.Empty);
+            context.ChangeTracker.Clear();
+
+            await _apiClient.Awaiting(a => a.Patch(
+                $"run/{created.Id}",
+                new()
+                {
+                    Body = new
+                    {
+                        RunType = nameof(RunType.Time),
+                        Info = "won't work",
+                    }
+                }
+            )).Should().ThrowAsync<RequestFailureException>().Where(e => e.Response.StatusCode == HttpStatusCode.Unauthorized);
+
+            Run? retrieved = await context.FindAsync<Run>(created.Id);
+            retrieved!.Info.Should().BeEmpty();
+        }
+
+        [TestCase(UserRole.Banned)]
+        [TestCase(UserRole.Registered)]
+        [TestCase(UserRole.Confirmed)]
+        public async Task UpdateRun_BadRole(UserRole role)
+        {
+            IServiceScope scope = _factory.Services.CreateScope();
+            IUserService users = scope.ServiceProvider.GetRequiredService<IUserService>();
+            ApplicationContext context = scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+
+            Run created = new()
+            {
+                CategoryId = _categoryId,
+                PlayedOn = LocalDate.MinIsoValue,
+                Time = Duration.FromSeconds(390),
+                UserId = TestInitCommonFields.Admin.Id,
+            };
+
+            context.Add(created);
+            await context.SaveChangesAsync();
+            created.Id.Should().NotBe(Guid.Empty);
+            context.ChangeTracker.Clear();
+
+            string email = $"testuser.updaterun.{role}@example.com";
+
+            RegisterRequest registerRequest = new()
+            {
+                Email = email,
+                Password = "Passw0rd",
+                Username = $"UpdateCatTest{role}"
+            };
+
+            await users.CreateUser(registerRequest);
+
+            LoginResponse res = await _apiClient.LoginUser(registerRequest.Email, registerRequest.Password);
+
+            User? user = await users.GetUserByEmail(email);
```

Comment:
This shouldn't really be necessary since we should have the user ID already after calling `CreateUser()`.